HealthMonitoring.Management.Core/Endpoint.cs

```diff
@@ -30,6 +30,18 @@ public void Dispose()
             UpdateLastModifiedTime();
         }
 
+        public void UpdateFirstRegistrationTime()
+        {
+            Metadata.SetFirstRegistrationTime(_timeCoordinator.UtcNow);
+            UpdateLastModifiedTime();
+        }
+
+        public void UpdateLastRegistrationUpdateTime()
+        {
+            Metadata.SetLastRegistrationUpdateTime(_timeCoordinator.UtcNow);
+            UpdateLastModifiedTime();
+        }
+
```

Comment:
Those methods should not be public and in fact they should not exists.
The FirstRegistrationTime could be passed to metadata when endpoint is created for a first time.
The LastRegistrationTime could be passed to metadata when endpoint is created for a first time or in UpdateMetadata().